packages/dev/core/src/AudioV2/abstractAudio/subProperties/abstractSpatialAudio.ts

```diff
@@ -0,0 +1,211 @@
+     */
+    spatialEnabled: boolean;
+    /**
+     * The maximum distance between the audio source and the listener, after which the volume is not reduced any further. Defaults to 10000.
+     * - This value is used only when the {@link spatialDistanceModel} is set to `"linear"`.
+     * @see {@link spatialDistanceModel}
+     */
+    spatialMaxDistance: number;
+    /**
+     * The spatial panning model. Defaults to "equalpower".
+     * - "equalpower" requires less CPU than "HRTF" but is less realistic for listeners with headphones or speakers close to the ears.
+     * - "HRTF" requires more CPU but is more realistic for listeners with headphones or speakers close to the ears.
+     */
+    spatialPanningModel: "equalpower" | "HRTF";
+    /**
+     * The spatial position. Defaults to (0, 0, 0).
+     */
+    spatialPosition: Vector3;
+    /**
+     * The distance for reducing volume as the audio source moves away from the listener – i.e. the distance the volume reduction starts at. Defaults to 1.
+     * - This value is used by all distance models.
+     * @see {@link spatialDistanceModel}
+     */
+    spatialReferenceDistance: number;
+    /**
+     * How quickly the volume is reduced as the source moves away from the listener. Defaults to 1.
+     * - This value is used by all distance models.
+     * @see {@link spatialDistanceModel}
+     */
+    spatialRolloffFactor: number;
+    /**
+     * The spatial rotation, as Euler angles. Defaults to (0, 0, 0).
+     */
+    spatialRotation: Vector3;
+    /**
+     * The spatial rotation, as a quaternion. Defaults to (0, 0, 0, 1).
+     */
+    spatialRotationQuaternion: Quaternion;
+}
+
+/**
+ * @param options The spatial audio options to check.
+ * @returns `true` if spatial audio options are defined, otherwise `false`.
+ */
+export function _HasSpatialAudioOptions(options: Partial<ISpatialAudioOptions>): boolean {
+    return (
+        options.spatialEnabled ||
+        options.spatialConeInnerAngle !== undefined ||
+        options.spatialConeOuterAngle !== undefined ||
+        options.spatialConeOuterVolume !== undefined ||
+        options.spatialDistanceModel !== undefined ||
+        options.spatialMaxDistance !== undefined ||
+        options.spatialPanningModel !== undefined ||
+        options.spatialPosition !== undefined ||
+        options.spatialReferenceDistance !== undefined ||
+        options.spatialRolloffFactor !== undefined ||
+        options.spatialRotation !== undefined ||
+        options.spatialRotationQuaternion !== undefined
+    );
+}
+
+/**
+ * Abstract class representing the `spatial` audio property on a sound or audio bus.
+ *
+ * @see {@link AudioEngineV2.listener}
+ */
+export abstract class AbstractSpatialAudio {
+    /**
+     * The spatial cone inner angle, in radians. Defaults to 2π.
+     * - When the listener is inside the cone inner angle, the volume is at its maximum.
+     */
+    public abstract get coneInnerAngle(): number;
```

Comment:
Is typescript complaining if you define this as `public coneInnerAngle: number` and then define getter/setter in the classes extending it?

Reply:
Same thing for the rest of the values in this class.

Reply:
TypeScript is fine with `public abstract coneInnerAngle: number;`